tests/langchain/test_langchain_databricks_dependency_extraction.py

```diff
@@ -134,11 +137,69 @@ def test_parsing_dependency_from_databricks_retriever(monkeypatch: pytest.Monkey
 
     monkeypatch.setitem(sys.modules, "databricks.vector_search.client", mock_module)
 
-    vectorstore = DatabricksVectorSearch(vs_index, text_column="content", embedding=embedding_model)
-    retriever = vectorstore.as_retriever()
-    resources = list(_extract_databricks_dependencies_from_retriever(retriever))
-    assert resources == [
-        DatabricksVectorSearchIndex(index_name="mlflow.rag.vs_index"),
+    # set embedding model
+    vectorstore_1 = DatabricksVectorSearch(
+        vs_index_1, text_column="content", embedding=embedding_model
+    )
+    retriever_1 = vectorstore_1.as_retriever()
+
+    # vs_index_2 has builtin embedding endpoint "embedding-model"
+    vectorstore_2 = DatabricksVectorSearch(vs_index_2, text_column="content")
+    retriever_2 = vectorstore_2.as_retriever()
+
+    from langchain.chat_models import ChatOpenAI
+
+    llm = ChatOpenAI(temperature=0)
+
+    assert list(_extract_databricks_dependencies_from_retriever(retriever_1)) == [
+        DatabricksVectorSearchIndex(index_name="mlflow.rag.vs_index_1"),
+        DatabricksServingEndpoint(endpoint_name="databricks-bge-large-en"),
+    ]
+
+    assert list(_extract_databricks_dependencies_from_retriever(retriever_2)) == [
+        DatabricksVectorSearchIndex(index_name="mlflow.rag.vs_index_2"),
+        DatabricksServingEndpoint(endpoint_name="embedding-model"),
+    ]
+
+    from langchain.retrievers.multi_query import MultiQueryRetriever
+
+    multi_query_retriever = MultiQueryRetriever.from_llm(retriever=retriever_1, llm=llm)
+    assert list(_extract_databricks_dependencies_from_retriever(multi_query_retriever)) == [
+        DatabricksVectorSearchIndex(index_name="mlflow.rag.vs_index_1"),
+        DatabricksServingEndpoint(endpoint_name="databricks-bge-large-en"),
+    ]
+
+    from langchain.retrievers import ContextualCompressionRetriever
+    from langchain.retrievers.document_compressors import LLMChainExtractor
+
+    compressor = LLMChainExtractor.from_llm(llm)
+    compression_retriever = ContextualCompressionRetriever(
+        base_compressor=compressor, base_retriever=retriever_1
+    )
+    assert list(_extract_databricks_dependencies_from_retriever(compression_retriever)) == [
+        DatabricksVectorSearchIndex(index_name="mlflow.rag.vs_index_1"),
+        DatabricksServingEndpoint(endpoint_name="databricks-bge-large-en"),
+    ]
```

Comment:
this appears to be using ChatOpenAI as llm. if we use ChatDatabricks in here, would we pick up that dependency too?

apologies if im missing something here.